### PR TITLE
Fixing `_read_config_file`

### LINKED ISSUE #41: Bug located in `_read_config_file` and `_migrate_config_file`.
### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

Infinite loop located between `_read_config_file` and `_migrate_config_file`. 

### 📝 Steps to reproduce

- Upload the branch from the PR https://github.com/ansys/pymapdl/pull/2054 which use the new `ansys-tools-path` version.
- Execute the following code:
 ```
from ansys.mapdl.core import launch_mapdl
mapdl = launch_mapdl()
```

This error appears:
```
  File "D:\repos\pyansys\pymapdl_1\.venv\lib\site-packages\ansys\tools\path\path.py", line 652, in _read_config_file
    _migrate_config_file(product_name)
  File "D:\repos\pyansys\pymapdl_1\.venv\lib\site-packages\ansys\tools\path\path.py", line 679, in _migrate_config_file
    new_config_data = _read_config_file(product_name)
  File "D:\repos\pyansys\pymapdl_1\.venv\lib\site-packages\ansys\tools\path\path.py", line 652, in _read_config_file
    _migrate_config_file(product_name)
  File "D:\repos\pyansys\pymapdl_1\.venv\lib\site-packages\ansys\tools\path\path.py", line 675, in _migrate_config_file
    with open(old_config_file) as f:
RecursionError: maximum recursion depth exceeded while calling a Python object
```

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

_No response_

### 🐍 Which Python version are you using?

3.10

### 📦 Installed packages

```shell
accessible-pygments==0.0.4
aiofiles==22.1.0
aiosqlite==0.19.0
alabaster==0.7.13
ansys-api-mapdl==0.5.1
ansys-api-platform-instancemanagement==1.0.0b3
ansys-dpf-core==0.8.1
ansys-dpf-gate==0.3.1
ansys-dpf-gatebin==0.3.1
ansys-grpc-dpf==0.7.1
-e git+https://github.com/pyansys/pymapdl.git@a4285ab4d22facc032b75df585a0450b5cfb4972#egg=ansys_mapdl_core
ansys-mapdl-reader==0.52.13
ansys-platform-instancemanagement==1.1.1
ansys-sphinx-theme==0.9.9
ansys-tools-path==0.2.1
anyio==3.6.2
appdirs==1.4.4
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
attrs==23.1.0
autopep8==2.0.2
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
cachetools==5.3.0
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
click==8.1.3
colorama==0.4.6
comm==0.1.3
contourpy==1.0.7
coverage==7.2.5
cycler==0.11.0
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
docutils==0.19
exceptiongroup==1.1.1
executing==1.2.0
fastjsonschema==2.16.3
fonttools==4.39.3
fqdn==1.5.1
geomdl==5.3.1
google-api-core==2.11.0
google-api-python-client==2.86.0
google-auth==2.17.3
google-auth-httplib2==0.1.0
googleapis-common-protos==1.59.0
grpcio==1.51.1
httplib2==0.22.0
idna==3.4
imageio==2.28.1
imageio-ffmpeg==0.4.8
imagesize==1.4.1
importlib-metadata==6.6.0
iniconfig==2.0.0
ipydatawidgets==4.3.3
ipykernel==6.23.0
ipython==8.13.2
ipython-genutils==0.2.0
ipywidgets==8.0.6
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
json5==0.9.11
jsonpointer==2.3
jsonschema==4.17.3
jupyter-events==0.6.3
jupyter-sphinx==0.4.0
jupyter-ydoc==0.2.4
jupyter_client==8.2.0
jupyter_core==5.3.0
jupyter_server==2.5.0
jupyter_server_fileid==0.9.0
jupyter_server_terminals==0.4.4
jupyter_server_ydoc==0.8.0
jupyterlab==3.6.3
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.7
jupyterlab_server==2.22.1
kiwisolver==1.4.4
livereload==2.6.3
MarkupSafe==2.1.2
matplotlib==3.7.1
matplotlib-inline==0.1.6
mistune==2.0.5
nbclassic==1.0.0
nbclient==0.7.4
nbconvert==7.4.0
nbformat==5.8.0
nest-asyncio==1.5.6
notebook==6.5.4
notebook_shim==0.2.3
numpy==1.24.3
numpydoc==1.5.0
packaging==23.1
pandas==2.0.1
pandocfilters==1.5.0
parso==0.8.3
pickleshare==0.7.5
Pillow==9.5.0
platformdirs==3.5.0
plotly==5.14.1
pluggy==1.0.0
pooch==1.7.0
prometheus-client==0.16.0
prompt-toolkit==3.0.38
protobuf==3.20.3
protoc-gen-swagger==0.1.0
psutil==5.9.5
pure-eval==0.2.2
pyansys-tools-report==0.5.0
pyansys-tools-versioning==0.3.3
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycodestyle==2.10.0
pycparser==2.21
pydata-sphinx-theme==0.13.3
Pygments==2.15.1
pyiges==0.2.1
pypandoc==1.11
pyparsing==3.0.9
pyrsistent==0.19.3
pytest==7.3.1
pytest-cov==4.0.0
pytest-rerunfailures==11.1.2
pytest-sphinx==0.5.0
python-dateutil==2.8.2
python-json-logger==2.0.7
pythreejs==2.4.2
pytz==2023.3
pyvista==0.39.1
pywin32==306
pywinpty==2.0.10
PyYAML==6.0
pyzmq==25.0.2
requests==2.30.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rsa==4.9
scipy==1.10.1
scooby==0.7.2
Send2Trash==1.8.2
six==1.16.0
sniffio==1.3.0
snowballstemmer==2.2.0
soupsieve==2.4.1
Sphinx==6.2.1
sphinx-autobuild==2021.3.14
sphinx-autodoc-typehints==1.23.0
sphinx-copybutton==0.5.2
sphinx-gallery==0.13.0
sphinx-notfound-page==0.8.3
sphinxcontrib-applehelp==1.0.4
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
sphinxcontrib-websupport==1.2.4
sphinxemoji==0.2.0
stack-data==0.6.2
tenacity==8.2.2
terminado==0.17.1
tinycss2==1.2.1
tomli==2.0.1
tornado==6.3.1
tqdm==4.65.0
traitlets==5.9.0
traittypes==0.2.1
typing_extensions==4.5.0
tzdata==2023.3
uri-template==1.2.0
uritemplate==4.1.1
urllib3==2.0.2
vtk==9.2.6
wcwidth==0.2.6
webcolors==1.13
webencodings==0.5.1
websocket-client==1.5.1
widgetsnbextension==4.0.7
y-py==0.5.9
ypy-websocket==0.8.2
zipp==3.15.0
```
```

### PR DESCRIPTION
**Pull request by clatapie**
Closes #41.

### CODE REVIEW

File: src/ansys/tools/path/path.py [hunk 1]
```diff
@@ -323,8 +323,6 @@ def find_mapdl(version=None, supported_versions=SUPPORTED_ANSYS_VERSIONS):
 def _find_installation(product: str, version=None, supported_versions=SUPPORTED_ANSYS_VERSIONS):
     if product == "mapdl":
         return find_mapdl(version, supported_versions)
-    elif product == "mechanical":
-        return find_mechanical(version, supported_versions)
```
Comment (germa89) [inline src/ansys/tools/path/path.py:327]: Why are you removing this?

File: src/ansys/tools/path/path.py [hunk 2]
```diff
@@ -651,7 +649,9 @@ def _clear_config_file() -> None:
 
 def _read_config_file(product_name: str) -> dict:
     """Read config file for a given product, migrating if needed"""
-    _migrate_config_file(product_name)
+
+    if not os.path.isfile(CONFIG_FILE):
+        _migrate_config_file(product_name)
```
Comment (germa89) [inline src/ansys/tools/path/path.py:654]: So I guess this is the key.

File: src/ansys/tools/path/path.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 from ansys.tools.path.misc import is_float, is_linux, is_windows
 
-LINUX_DEFAULT_DIRS = [["/", "usr", "ansys_inc"], ["/", "ansys_inc"], ["/", "install", "ansys_inc"]]
+LINUX_DEFAULT_DIRS = [["/", "usr", "ansys_inc"], ["/", "ansys_inc"]]
```
Comment (germa89) [inline src/ansys/tools/path/path.py:13]: why are you removing this? I see it is not a common path. Have you check who added it? and why?

File: tests/test_path.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def test_migration():
     _clear_config_file()
 
     old_settings_dir = appdirs.user_data_dir(f"ansys_mapdl_core")
-    os.makedirs(old_settings_dir)
+    os.makedirs(old_settings_dir, exist_ok=True)
```
Comment (germa89) [inline tests/test_path.py:53]: Very good!

**Review by germa89 — CHANGES_REQUESTED** [review]
Please address my questions and we can merge after.

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**